kms/api/CloudKmsTest/CloudKmsTest.cs

```diff
@@ -279,6 +294,124 @@ private bool AssertRoleAndMemberPresent(string response, string role, string mem
             }
             return (foundRole && foundMember);
         }
+
+        private CryptoKeyVersionName BuildAsymmetricKey(KeyRingName keyRingName, String cryptoKey,
+                CryptoKey.Types.CryptoKeyPurpose purpose,
+                CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm algorithm)
+        {
+            KeyManagementServiceClient client = KeyManagementServiceClient.Create();
+            CryptoKey cryptoKeyToCreate = new CryptoKey
+            {
+                Purpose = purpose,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = algorithm,
+                    ProtectionLevel = ProtectionLevel.Software,
+                },
+            };
+            CryptoKey key = client.CreateCryptoKey(keyRingName, cryptoKey, cryptoKeyToCreate);
+
+            CryptoKeyVersionName ckvName = CryptoKeyVersionName.Parse(key.Name + "/cryptoKeyVersions/1");
+            CryptoKeyVersion ckv = client.GetCryptoKeyVersion(ckvName);
+
+            // The version is generated asynchronously. Wait for it to be in state Enabled before
+            // proceeding. Throw an exception if it hasn't been generated after 5 minutes.
+            Task refresher = Task.Run(async () =>
+            {
+                while (ckv.State != CryptoKeyVersion.Types.CryptoKeyVersionState.Enabled)
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(5));
+                    ckv = client.GetCryptoKeyVersion(ckvName);
+                }
+            });
+            if (!refresher.Wait(TimeSpan.FromMinutes(5)))
+            {
+                throw new TimeoutException($"Timed out waiting for version '{{ckv.Name}}' to become enabled.");
+            }
+
+            return ckvName;
+        }
+
+        [Fact]
+        public void TestAsymmetricSignAndVerify()
+        {
+            string contentFile = @"../../../../test/data/test_file.txt";
+            string signatureFile = @"../../../../test/data/test_file_signature.bin";
+            string timeStamp = $"-{DateTime.Now.ToString("yyyyMMddHHmmssfff")}";
+            string keyRing = $"testKeyRing{timeStamp}";
+            string cryptoKey = $"testCryptoKey{timeStamp}";
+            KeyRingName keyRingName = new KeyRingName(s_projectId, "global", keyRing);
+
+            Run("createKeyRing", keyRingName.ProjectId, keyRingName.LocationId, keyRingName.KeyRingId);
+
+            CryptoKeyVersionName version = BuildAsymmetricKey(
+                keyRingName, cryptoKey, CryptoKey.Types.CryptoKeyPurpose.AsymmetricSign,

```

Comment:
nit: Same when passing parameters.

Reply:
As above, this is consistent with the code that already exists in this file.

Reply:
As above.